### ISSUE TITLE
8 tests fail in my environment

### ISSUE THREAD

**Issue by NelsonMinar** [openaddresses/machine#312]
Haven't run the test suite in a long time. When I do, 8 tests fail. They all seem related to the CI package; perhaps those tests require some online server credentials I don't have? I'm testing in Ubuntu with Python 2, both in a virtualenv and a known-working environment. I tested with `b9b2559`, a recent version right before some changes I made.

[Full transcript attached](https://github.com/openaddresses/machine/files/163675/errors.txt), the tests that fail are:

```
ERROR: test_is_merged_to_master (openaddr.tests.ci.TestBatch)
ERROR: test_download_palmbeach (openaddr.tests.cache.TestCacheEsriDownload)
ERROR: test_webhook_badjson_content (openaddr.tests.ci.TestHook)
ERROR: test_webhook_one_master_commit (openaddr.tests.ci.TestHook)
ERROR: test_webhook_two_branch_commits (openaddr.tests.ci.TestHook)
ERROR: test_webhook_two_master_commits_early_fail (openaddr.tests.ci.TestHook)
ERROR: test_webhook_two_master_commits_late_fail (openaddr.tests.ci.TestHook)
ERROR: test_iterate_local_processed_files (openaddr.tests.TestPackage)
```

(A secondary problem is that there's warning and error logspam even when the tests pass, which makes the test output hard to follow. When I was actively working on the test suite I was doing my best to make sure that the test path never logged above info level, but maybe that's a mistake. Perhaps we should configure the test runner to simply print no logging messages at all.)

**Comment by migurski** [comment]
Looks like your test errors are due to missing SNS configuration. I’ll make sure this works by default.

**Comment by migurski** [comment]
On closer look, I’m not sure I can actually tell what’s going on here. The errors seem to arise from Flask test_client, and the SNS configuration is just a red herring. Can you say more about your setup?

**Comment by NelsonMinar** [comment]
It's just a garden-variety Ubuntu 14.04 system. Tests fail the same way in a clean virtualenv (ie: new versions of stuff) and my crufty old dev system. I don't think I have any special S3 or Amazon stuff in my environment, but it's possible I do. Any idea what I should check for? I did have a `.boto` file but moving that out of the way doesn't fix anything.

Are the CI tests talking to a live service?

**Comment by migurski** [comment]
Hm that matches expectations for production as well, though I recall you saying that you had to engage in some funny business to binary packages installed.

The CI tests don't talk to a live service. Each errors complains about not having SNS configured, but that's only because it's trying to send an error report for a part of the tests that don't normally lead to an error. I might be able to mask those with some fake SNS credentials, but it wouldn't fix the failures.

I guess my next step would be set up an Ubuntu 14.04 environment and try this out.

**Comment by migurski** [comment]
I tried running through the complete process on a fresh Ubuntu machine, and could not replicate your 8 test failures. Here's what I did:

```SHELL
sudo apt-get update -y
sudo apt-get upgrade -y
sudo apt-get install git -y

git clone https://github.com/openaddresses/machine.git && cd machine

# Packages mostly from contents of chef recipes
sudo apt-get install python-cairo python-gdal python-pip -y
sudo apt-get install python-dev libpq-dev memcached postgresql -y

sudo pip install virtualenv
virtualenv  --system-site-packages -p `which python2` .venv
source .venv/bin/activate

pip install -r requirements.txt

sudo -u postgres psql -c "create user dashboard password ''"
sudo -u postgres psql -c "create user openaddr password 'openaddr'"
sudo -u postgres psql -c "create database openaddr owner openaddr"

# Borrowed env from CI config in https://github.com/openaddresses/machine/blob/master/circle.yml#L5-L6
env BOTO_CONFIG=/tmp/nowhere DATABASE_URL=postgres://openaddr:openaddr@localhost/openaddr python test.py
```

**Comment by NelsonMinar** [comment]
Thanks for doing this thorough examination. Your success suggests the problem is my environment, not the code. I tried tinkering for awhile and couldn't figure out how to fix it or even what the source of the errors is, will need to try again with more patience.

The big news in your transcript above is the need for postgres. I got that set up the way you have it, but still see the same 8 errors.

Including system-site-packages in the venv makes it harder to have a clean setup, but that's the only way I know to make GDAL work on Ubuntu 14.04.

When you do `pip install -r requirements.txt`, that creates a copy of the source code in the venv site-packages, right? What do you do in your development environment when you're editing code? Do you rerun pip every time, or do you install with `pip install -e` or something?

I'm mystified that my transcript includes logging messages. I re-read the test.py logging code and it explicitly sets it up to only log CRITICAL messages (ie: nothing). Do the CI tests fork new Python processes?

**Comment by migurski** [comment]
You must have had Postgres set up previously, though, or else you would have seen significantly more errors?

I don't do anything special in my dev environment when editing code. Even though pip installs a copy of the OpenAddresses code into the venv, that copy will be ignored if it can find one in the local directory thanks to `''` at the front of `sys.path`:

```
>>> import openaddr
>>> openaddr
<module 'openaddr' from 'openaddr/__init__.pyc'>
```

So, no need to use `pip install -e` or anything of the sort.

The logging messages being present are an unfortunate bummer resulting from log configuration internal to the CI Flask application that's executed as part of the tests.

**Comment by migurski** [comment]
Any interest in trying this again? Otherwise I’ll close out the issue.

**Comment by NelsonMinar** [comment]
I'll close it, since it seems to be just me. Next time I work on the code I'll rebuild my dev environment.
